python/semantic_kernel/connectors/ai/ai_response.py

```diff
@@ -0,0 +1,19 @@
+from typing import Any, AsyncGenerator
+
+from semantic_kernel.connectors.ai.ai_request_settings import AIRequestSettings
+from semantic_kernel.sk_pydantic import SKBaseModel
+
+
+class AIResponse(SKBaseModel):
+    raw_response: Any
+    request_settings: AIRequestSettings
+
+    @property
+    def content(self) -> Any:
+        """Get the content of the response."""
+        return self.raw_response
+
+    @property
+    async def streaming_content(self) -> AsyncGenerator[Any, None]:
```

Comment:
How would this work if there are models/responses that don't support streaming?

Reply:
Looking down further, it seems the error handling is per method/type. I see.

Reply:
yeah, simple integrations could just put in the raw_response and then either return that (non-stream) or yield (stream) that string.